### PR TITLE
link: always include requested symbol in kprobe creation errors

### LINKED ISSUE #956: Improve error logs when retrying to attach a kprobe with an arch specific prefix
The PR https://github.com/cilium/ebpf/pull/304 added a mechanism to retry on a kprobe symbol in case it was not found, typically adding an arch-specific prefix to the symbol. It's a good idea since it allows architecture-independent symbols for syscalls while on the downside creating an implicit alias for some symbols (on amd64 you end up having `cilium/ebpf` "exposing" `__x64_sys_listen` and `sys_listen` that target the same `__x64_sys_listen` symbol).

However, the error message is sometimes a bit puzzling because you wonder when trying `__x64_sys_listen` on arm64, you end up with an error like:
```
creating perf_kprobe PMU: token __arm64___x64_sys_listen: not found: no such file or directory
```

Maybe it would be possible to include in the error message that `__x64_sys_listen` was first tried (which was the user input) and then `__arm64___x64_sys_listen`, which could be the arch-specific symbol the user was referring to.

Related to https://github.com/cilium/ebpf/issues/266.

### PR DESCRIPTION
**Pull request by ti-mo**
Before, when a caller would try to set a kprobe on e.g. __x64_sys_listen, the error message would look something like 'creating perf_kprobe PMU: token __arm64___x64_sys_listen: not found: no such file or directory'. This doesn't mention that the requested symbol has been attempted in addition to the prefixed symbol mentioned in the error message.

If a symbol doesn't exist, which is going to be the vast majority of cases, the caller will only see the prefixed symbol echoed in the error message, which can make them wonder if the original symbol has been attempted at all.

This patch makes PMU and tracefs errors more uniform and always mentions the requested symbol in the error string, which only used to happen for tracefs.

Fixes #956

### CODE REVIEW

**Review by mtardy — APPROVED** [review]
Thanks, that would be lovely! 😸

So now it will look like this:
```text
creating perf_kprobe PMU (tried '_x64_sys_listen'): token __arm64___x64_sys_listen: not found: no such file or directory
```
It still requires knowing that the lib is retrying on fail with an arch-specific prefix but at least the original symbol appears in the error which is excellent!

File: link/kprobe.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func kprobe(symbol string, prog *ebpf.Program, opts *KprobeOptions, ret bool) (*
 		return tp, nil
 	}
 	if err != nil && !errors.Is(err, ErrNotSupported) {
-		return nil, fmt.Errorf("creating perf_kprobe PMU: %w", err)
+		return nil, fmt.Errorf("creating perf_kprobe PMU (tried '%s'): %w", symbol, err)
```
Comment (lmb) [inline link/kprobe.go:202]: Maybe it's better to just return the original error?

```go
origErr := pmuKprobe(...)

fallbackErr := pmuKprobe(...)
if fallbackErr != nil { return origErr }
```

Comment (mtardy) [reply link/kprobe.go:202]: The best would be both, being explicit about the fact that symbol1 and adaptedSymbol1 failed. Losing the fallbackErr could make the behavior not very clear I think.

**Review by mmat11 — APPROVED** [review]

File: link/kprobe.go [hunk 2]
```diff
@@ -199,7 +207,7 @@ func kprobe(symbol string, prog *ebpf.Program, opts *KprobeOptions, ret bool) (*
...
```
Comment (lmb) [inline link/kprobe.go:210]:
```suggestion
		return nil, fmt.Errorf("creating perf_kprobe PMU (arch specific fallback for %q): %w", symbol, err)
```

**Review by lmb — APPROVED** [review]

**Pull request merged**